pkg/framework/plugins/defaultevictorfilter/defaultevictorfilter.go

```diff
@@ -0,0 +1,203 @@
+package defaultevictorfilter
+
+import (
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/descheduler/pkg/apis/componentconfig"
+	nodeutil "sigs.k8s.io/descheduler/pkg/descheduler/node"
+	podutil "sigs.k8s.io/descheduler/pkg/descheduler/pod"
+	"sigs.k8s.io/descheduler/pkg/framework"
+	"sigs.k8s.io/descheduler/pkg/utils"
+)
+
+const (
+	PluginName = "IsEvictableFilter"
+	evictPodAnnotationKey = "descheduler.alpha.kubernetes.io/evict"
+)
+
+
+type constraint func(pod *v1.Pod) error
+
+
+type IsEvictableFilter struct {
+	args      *componentconfig.IsEvictableFilterArgs
+	constraints []constraint
+}
+
+// WithPriorityThreshold sets a threshold for pod's priority class.
+// Any pod whose priority class is lower is evictable.
+func WithPriorityThreshold(priority int32) func(opts *FilterOptions) {
```

Comment:
All the `With*` options can be passed into the default evictor plugin through arguments in IsEvictableFilterArgs

Reply:
You mean

```
type IsEvictableFilterArgs struct {
	metav1.TypeMeta

	Nodes                   []*v1.Node
	EvictLocalStoragePods   bool
	EvictSystemCriticalPods bool
	IgnorePvcPods           bool
	EvictFailedBarePods     bool
}
```

should also have 

```
opts                    []func(opts *defaultevictorfilter.FilterOptions)
```
?

The problem here is that importing defaultevictorfilter in types_pluginargs.go will result in a import cycle. How should I go about that?

Also, is array of functions supported by the deepcopy generation of code?

Reply:
Should I move all `With*` options back to evictions.go?

Reply:
We cant use slice of functions in IsEvictableFilterArgs because it is not supported by deepcopy code generation:

```
F0831 15:16:09.615636  527719 deepcopy.go:816] Hit an unsupported type func(opts *./pkg/apis/componentconfig.FilterOptions) for []func(opts *./pkg/apis/componentconfig.FilterOptions)
goroutine 1 [running]:
k8s.io/klog/v2.stacks(0x1)
        /home/lucas/workspace/descheduler/_tmp/kube-verify.8R6o0i/descheduler/vendor/k8s.io/klog/v2/klog.go:860 +0x89
k8s.io/klog/v2.(*loggingT).output(0x9f9700, 0x3, 0x0, 0xc00b986e70, 0x1, {0x87e7ed?, 0x2?}, 0xc000600400?, 0x0)
        /home/lucas/workspace/descheduler/_tmp/kube-verify.8R6o0i/descheduler/vendor/k8s.io/klog/v2/klog.go:825 +0x685
k8s.io/klog/v2.(*loggingT).printfDepth(0x9f9700, 0x0?, 0x0, {0x0, 0x0}, 0xc00befcd50?, {0x79c37d, 0x21}, {0xc00bf04060, 0x2, ...})
        /home/lucas/workspace/descheduler/_tmp/kube-verify.8R6o0i/descheduler/vendor/k8s.io/klog/v2/klog.go:630 +0x1f2
k8s.io/klog/v2.(*loggingT).printf(...)
        /home/lucas/workspace/descheduler/_tmp/kube-verify.8R6o0i/descheduler/vendor/k8s.io/klog/v2/klog.go:612
k8s.io/klog/v2.Fatalf(...)
        /home/lucas/workspace/descheduler/_tmp/kube-verify.8R6o0i/descheduler/vendor/k8s.io/klog/v2/klog.go:1516
k8s.io/gengo/examples/deepcopy-gen/generators.(*genDeepCopy).doSlice(0xc00befd260?, 0xc00befd290?, 0x1?)
        /home/lucas/workspace/descheduler/_tmp/kube-verify.8R6o0i/descheduler/vendor/k8s.io/gengo/examples/deepcopy-gen/generators/deepcopy.go:816 +0x506
k8s.io/gengo/examples/deepcopy-gen/generators.(*genDeepCopy).generateFor(0xc00bea1270?, 0x79edd1?, 0x25?)
        /home/lucas/workspace/descheduler/_tmp/kube-verify.8R6o0i/descheduler/vendor/k8s.io/gengo/examples/deepcopy-gen/generators/deepcopy.go:698 +0x403
k8s.io/gengo/examples/deepcopy-gen/generators.(*genDeepCopy).doStruct(0xc00bee5230?, 0x7775f1?, 0x8?)
        /home/lucas/workspace/descheduler/_tmp/kube-verify.8R6o0i/descheduler/vendor/k8s.io/gengo/examples/deepcopy-gen/generators/deepcopy.go:868 +0x46c
k8s.io/gengo/examples/deepcopy-gen/generators.(*genDeepCopy).generateFor(0xc00bea1270?, 0x7a4f2b?, 0x38?)
        /home/lucas/workspace/descheduler/_tmp/kube-verify.8R6o0i/descheduler/vendor/k8s.io/gengo/examples/deepcopy-gen/generators/deepcopy.go:698 +0x403
k8s.io/gengo/examples/deepcopy-gen/generators.(*genDeepCopy).GenerateType(0xf0?, 0xc00bddfa80, 0xc00a251ec0, {0x806248?, 0xc00bed5ee0})
        /home/lucas/workspace/descheduler/_tmp/kube-verify.8R6o0i/descheduler/vendor/k8s.io/gengo/examples/deepcopy-gen/generators/deepcopy.go:611 +0x765
k8s.io/gengo/generator.(*Context).executeBody(0xc00bddfa80, {0x805a68?, 0xc00b1242a0}, {0x80a2d8, 0xc00bddf880})
        /home/lucas/workspace/descheduler/_tmp/kube-verify.8R6o0i/descheduler/vendor/k8s.io/gengo/generator/execute.go:321 +0x131
k8s.io/gengo/generator.(*Context).ExecutePackage(0xc00a131b00, {0x773b13, 0x2}, {0x809350, 0xc00b102e10})
        /home/lucas/workspace/descheduler/_tmp/kube-verify.8R6o0i/descheduler/vendor/k8s.io/gengo/generator/execute.go:282 +0xfbd
k8s.io/gengo/generator.(*Context).ExecutePackages(0xc00a131b00?, {0x773b13, 0x2}, {0xc00b1cca00?, 0x4, 0xc0000ddd78?})
        /home/lucas/workspace/descheduler/_tmp/kube-verify.8R6o0i/descheduler/vendor/k8s.io/gengo/generator/execute.go:51 +0x1de
k8s.io/gengo/args.(*GeneratorArgs).Execute(0xc0000e8420, 0xc0000dde30?, {0x775a60, 0x6}, 0x7a8298)
        /home/lucas/workspace/descheduler/_tmp/kube-verify.8R6o0i/descheduler/vendor/k8s.io/gengo/args/args.go:213 +0x1c8
main.main()
        /home/lucas/workspace/descheduler/_tmp/kube-verify.8R6o0i/descheduler/vendor/k8s.io/code-generator/cmd/deepcopy-gen/main.go:75 +0x385
```

That's why I chose to leave NodeIndexer and Opts  out

Reply:
Apologies for not being more clear. I meant:
- WithPriorityThreshold -> new PriorityThresholds field in `DefaultEvictorFilterArgs`
- WithNodeFit -> new NodeFit field in `DefaultEvictorFilterArgs`
...